app/src/main/java/com/example/tmdb/screens/home/HomeViewModel.kt

```diff
@@ -0,0 +1,203 @@
+package com.example.tmdb.screens.home
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import androidx.paging.filter
+import com.example.tmdb.model.*
+import com.example.tmdb.repository.MoviesRepository
+import com.example.tmdb.repository.TvRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    private val moviesRepository: MoviesRepository,
+    private val tvRepository: TvRepository,
+) : ViewModel(){
+
+    // Movie
+    private var _trendingMoviesDay = mutableStateOf <Flow<PagingData<Movie>>>(emptyFlow())
+    val trendingMoviesDay: State<Flow<PagingData<Movie>>> = _trendingMoviesDay
+
+    private var _trendingMoviesWeek = mutableStateOf <Flow<PagingData<Movie>>>(emptyFlow())
+    val trendingMoviesWeek: State<Flow<PagingData<Movie>>> = _trendingMoviesWeek
+
+    private val _upcomingMovies = mutableStateOf<Flow<PagingData<Movie>>>(emptyFlow())
+    val upcomingMovies: State<Flow<PagingData<Movie>>> = _upcomingMovies
+
+    private val _topRatedMovies = mutableStateOf<Flow<PagingData<Movie>>>(emptyFlow())
+    val topRatedMovies: State<Flow<PagingData<Movie>>> = _topRatedMovies
+
+    private val _nowPlayingMovies = mutableStateOf<Flow<PagingData<Movie>>>(emptyFlow())
+    val nowPlayingMovies: State<Flow<PagingData<Movie>>> = _nowPlayingMovies
+
+    private val _popularMovies = mutableStateOf<Flow<PagingData<Movie>>>(emptyFlow())
+    val popularMovies: State<Flow<PagingData<Movie>>> = _popularMovies
+
+    //Tv
+    private val _trendingTv = mutableStateOf<Flow<PagingData<Tv>>>(emptyFlow())
+    val trendingTv: State<Flow<PagingData<Tv>>> = _trendingTv
+
+    private val _onAirTv = mutableStateOf<Flow<PagingData<Tv>>>(emptyFlow())
+    val onAirTv: State<Flow<PagingData<Tv>>> = _onAirTv
+
+    private val _airingTodayTv = mutableStateOf<Flow<PagingData<Tv>>>(emptyFlow())
+    val airingTodayTv: State<Flow<PagingData<Tv>>> = _airingTodayTv
+
+    private val _popularTv = mutableStateOf<Flow<PagingData<Tv>>>(emptyFlow())
+    val popularTv: State<Flow<PagingData<Tv>>> = _popularTv
+
+    init{
+        getTrendingMoviesDay(null)
+        getTrendingMoviesWeek(null)
+        getNowPayingMovies(null)
+        getUpcomingMovies(null)
+        getPopularMovies(null)
+
+        getAiringTodayTv(null)
+        getTrendingTv(null)
+        getOnTheAirTv(null)
+        getOnTheAirTv(null)
+        getPopularTv(null)
+    }
+
+    //Movie
+    fun getTrendingMoviesDay(genreId: Int?){
+        viewModelScope.launch {
+            _trendingMoviesDay.value = if(genreId != null){
```

Comment:
I don't understand this if clause.

You are not passing genreId anywhere in the code.

Reply:
Ah, seems I forgot to remove this, since I was testing if filtering through genres would work. I removed it in my newest commit.